src/general/index.ts

```diff
@@ -110,3 +110,121 @@ export function getSelectedTabIndex(tabs: any[], selectedTabName: string): numbe
+ *  <li> <span style="font-weight: bold;">type</span> => MIME-TYPE of the file. 'application/pdf', 'application/gzip', 'image/png' </li>
+ *  <li> <span style="font-weight: bold;">fileName</span> => Expected name of the downloaded file </li>
+ *  <li> <span style="font-weight: bold;">downloadMethod</span> => 'blob' or 'url' </li>
+ *  <li> <span style="font-weight: bold;">fileExtension</span> => Expected extension of the downloaded file </li>
+ *  <li> <span style="font-weight: bold;">fileUrl</span> => downloadable file's url. Required if downloadMethod is 'url' </li>
+ * </ul>
+ *
+ * @example
+ * ```
+ * downloadFile({
+ *  file: fileBlobObject,
+ *  type: 'application/pdf',
+ *  fileName: 'MyFile',
+ *  fileExtension: 'pdf',
+ *  downloadMethod: 'blob',
+ *  fileUrl: null
+ * }) // *Downloads file of type PDF on the client's machine named MyFile.pdf*
+ * ```
+ *
+ * @category General Method
+ */
+export function downloadFile(downloadConfig:{ file:File | null; type:string; fileName:string; downloadMethod:string; fileExtension:string; fileUrl:string | null }) {
+
+  const DOWNLOAD_FILE_METHOD = {
+    BLOB: 'blob',
+    URL: 'url'
+  };
+
+  const { file = null, type, fileName, downloadMethod = DOWNLOAD_FILE_METHOD.URL, fileExtension, fileUrl } = downloadConfig;
+
+
+  const createFileUrlFromBlob = (file: File | null, type:string) => {
+    if (file) {
+      // It is necessary to create a new blob object with mime-type explicitly set
+      // otherwise only Chrome works like it should
+      const newBlob = new Blob([ file ], { type });
+
+      // For other browsers:
+      // Create a link pointing to the ObjectURL containing the blob.
+      const fileURL = (window.URL || window.webkitURL).createObjectURL(newBlob);
+
+      return fileURL;
+
+    } else {
+      throw new Error('file/blob is null');
+    }
+  };
+
+
+  const downloadFileFromUrl = (fileUrl:string | null, fileName:string, extension:string) => {
+    if (fileUrl) {
+      const link = document.createElement('a');
+
+      link.href = fileUrl;
+      link.download = `${fileName}.${extension}`;
+      link.target = '_blank';
+
+      document.body.appendChild(link);
+      link.click();
+
+      setTimeout(function() {
+        document.body.removeChild(link);
+        window.URL.revokeObjectURL(fileUrl);
+      }, 10);
+
+    } else {
+      throw new Error('fileUrl is empty');
+    }
+  };
+
+
+  try {
+    if (typeof window === 'undefined') {
+      throw new Error('window is undefined');
+    }
+
+    switch (downloadMethod) {
+      case DOWNLOAD_FILE_METHOD.BLOB:
+        const fileUrlFromBlob = createFileUrlFromBlob(file, type);
+
+        downloadFileFromUrl(fileUrlFromBlob, fileName, fileExtension);
+        break;
+
+      case DOWNLOAD_FILE_METHOD.URL:
+        downloadFileFromUrl(fileUrl, fileName, fileExtension);
+
+        break;
+    }
+
+  } catch (err) {
+    console.error('File download failed - ', err);
```

Comment:
Maybe not today, but later we might have to throw an error as calling function might want to show some UI on download failure.

Reply:
Can start throwing an error from here as well.. Regardless function toh phatt hi gaya.. Agar catch se throw kardunga dubara toh ek level upar bubble ho jaega.. Ideally inn methods ki implementation bhi try..catch mein wrapped honi chahie.

Reply:
Will fix this